node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -101,8 +102,9 @@ case class DataMessageHandler(
           Try {
             val blockFilter =
               BlockFilter.fromBytes(filter.filterBytes, filter.blockHash)
-            callbacks.onCompactFilterReceived.foreach(
-              _.apply(filter.blockHash, blockFilter))
+            callbacks.onCompactFilterReceived.foldLeft(FutureUtil.unit)(
```

Comment:
I think we need all types of callback to be called synchronously, i.e. plugged into the `for` statements, like you did in line 202 for `onBlockHeadersReceived`.